Tanga/Features/Authentication/AuthManager+DeleteAccount.swift

```diff
@@ -0,0 +1,268 @@
+//
+//  AuthManager+DeleteAccount.swift
+//  Tanga
+//
+//  Created by Rygel Louv on 11/02/2025.
+//
+
+import AuthenticationServices
+import FirebaseAuth
+import GoogleSignIn
+import OSLog
+
+// MARK: - Protocols
+
+/// Protocol defining the core authentication management functionality
+protocol AccountDeletionManaging {
+    /// Deletes the current user's account and revokes all associated provider access
+    /// - Throws: AuthError if deletion fails or user is not found
+    func deleteUserAccount() async throws
+    
+    /// Reauthenticates the user if their last sign-in is too old
+    /// - Parameter user: The Firebase User to reauthenticate
+    /// - Throws: AuthError if reauthentication fails
+    func reauthenticateIfNeeded(_ user: FirebaseAuth.User) async throws
+}
+
+/// Protocol for handling Apple ID account deletion and access revocation
+protocol AppleAccountTerminating {
+    /// Reauthenticates a user using their Apple ID credentials for account deletion
+    /// - Parameters:
+    ///   - user: The Firebase User to reauthenticate
+    ///   - credential: The Apple ID credential to use for reauthentication
+    /// - Throws: AuthError if reauthentication fails or credentials are invalid
+    func reauthenticate(user: FirebaseAuth.User, credential: ASAuthorizationAppleIDCredential) async throws
+    
+    /// Revokes the app's access to the user's Apple ID and prepares for account deletion
+    /// - Parameter credential: The Apple ID credential to revoke
+    /// - Throws: AuthError if revocation fails
+    func revokeAccess(credential: ASAuthorizationAppleIDCredential) async throws
+}
+
+/// Protocol for handling Google account deletion and access revocation
+protocol GoogleAccountTerminating {
+    /// Reauthenticates a user using their Google account for deletion
+    /// - Parameter user: The Firebase User to reauthenticate
+    /// - Throws: AuthError if reauthentication fails
+    func reauthenticate(user: FirebaseAuth.User) async throws
+    
+    /// Revokes the app's access to the user's Google account and prepares for deletion
+    /// - Throws: AuthError if revocation fails
+    func revokeAccess() async throws
+}
+
+// MARK: Auth Manager Account Deletion
+
+// Inspired from https://medium.com/firebase-developers/deleting-user-account-revoke-access-token-0e30d7a351bb
+extension AuthManager: AccountDeletionManaging {
+    
+    // MARK: - Public Methods
+        
+    func deleteUserAccount() async throws {
+        print("delete user account")
+        // Verify user exists and get their last sign-in date
+        guard let user = Auth.auth().currentUser,
+              let lastSignInDate = user.metadata.lastSignInDate else {
+            throw AuthError.userNotFound
+        }
+        
+        let providers = user.providerData.map { $0.providerID }
```

Comment:
You can directly use the keypath here
```suggestion
        let providers = user.providerData.map(\.providerID)
```